src/mlpack/methods/softmax_regression/softmax_regression_function.hpp

```diff
@@ -38,12 +37,25 @@ class SoftmaxRegressionFunction
   const arma::mat InitializeWeights();
 
   /**
+   * Initialize Softmax Regression weights(trainable parameters) with
+   * the given parameters.
+   * @param featureSize The features size of the training set
+   * @param numClasses Number of classes for classification.
+   * @param fitIntercept Intercept term flag.
+   * @return weights after initialize
+   */
+  static const arma::mat InitializeWeights(const size_t featureSize,
+                                           const size_t numClasses,
+                                           const bool fitIntercept = false);
```

Comment:
My only suggestion here might be to take the return value as a parameter (i.e. `arma::mat& weights`).  RVO might successfully optimize out the copy in some cases, but I'm not sure that every compiler will do RVO correctly in this case.